icu4j/main/core/src/main/java/com/ibm/icu/text/NFRule.java

```diff
@@ -1149,22 +1149,27 @@ private Number matchToDelimiter(String text, int startPos, double baseVal,
         // instance of "delimiter" we just found.
         if (!allIgnorable(delimiter)) {
             ParsePosition tempPP = new ParsePosition(0);
-            Number tempResult;
-
-            // use findText() to search for "delimiter".  It returns a two-
-            // element array: element 0 is the position of the match, and
-            // element 1 is the number of characters that matched
-            // "delimiter".
-            int[] temp = findText(text, delimiter, pluralFormatDelimiter, startPos);
-            int dPos = temp[0];
-            int dLen = temp[1];
-
-            // if findText() succeeded, isolate the text preceding the
-            // match, and use "sub" to match that text
-            while (dPos >= 0) {
+            Number bestResult = null;
+            int currPos = startPos;
+
+            for (;;) {
+                // use findText() to search for "delimiter". It returns a two-
+                // element array: element 0 is the position of the match, and
+                // element 1 is the number of characters that matched
+                // "delimiter".
+                tempPP.setIndex(0);
+                int[] temp = findText(text, delimiter, pluralFormatDelimiter, currPos);
+                int dPos = temp[0];
+                int dLen = temp[1];
+
+                if (dPos < 0) {
+                    break;
+                }
+                // if findText() succeeded, isolate the text preceding the
+                // match, and use "sub" to match that text
```

Comment:
Changes to this method will now allow "10.000.000." to be parsed correctly as an ordinal in languages that use the period as a grouping separator and an ordinal indicator. The value parsed is now 10000000 instead of 10.